pixivpy3/api.py

```diff
@@ -144,28 +183,43 @@ def auth(self, username=None, password=None, refresh_token=None, headers=None):
             self.access_token = token.response.access_token
             self.refresh_token = token.response.refresh_token
         except json.JSONDecodeError:
-            raise PixivError('Get access_token error! Response: %s' % token, header=r.headers, body=r.text)
+            raise PixivError(
+                "Get access_token error! Response: %s" % token,
+                header=r.headers,
+                body=r.text,
+            )
 
         # return auth/token response
         return token
 
-    def download(self, url, prefix='', path=os.path.curdir, name=None, replace=False, fname=None,
-                 referer='https://app-api.pixiv.net/'):
+    def download(
+        self,
+        url,
+        prefix="",
+        path=os.path.curdir,
+        name=None,
+        replace=False,
+        fname=None,
+        referer="https://app-api.pixiv.net/",
+    ):
+        # type: (str, str, str, Optional[str], bool, Optional[Union[str, IO[bytes]]], str) -> bool
         """Download image to file (use 6.0 app-api)"""
-        if hasattr(fname, 'write'):
+        if isinstance(fname, io.IOBase):
             # A file-like object has been provided.
             file = fname
         else:
             # Determine file path by parameters.
-            name = prefix + (name or fname or os.path.basename(url))
+            name = prefix + str(name or fname or os.path.basename(url))
             file = os.path.join(path, name)
             if os.path.exists(file) and not replace:
                 return False
 
-        with self.requests_call('GET', url, headers={'Referer': referer}, stream=True) as response:
+        with self.requests_call(
+            "GET", url, headers={"Referer": referer}, stream=True
+        ) as response:
             if isinstance(file, str):
-                with open(file, 'wb') as out_file:
+                with open(file, "wb") as out_file:
                     shutil.copyfileobj(response.raw, out_file)
-            else:
+            elif isinstance(fname, io.IOBase):
```

Comment:
Why are we checking `fname` again here?

Reply:
fixed: [3bf111c](https://github.com/upbit/pixivpy/pull/206/commits/3bf111c8c946d931c1d0fb95635f8c9e6fc9dcd7)